### PR TITLE
Add .NET 6 templates support, VS2022, Reunion 0.8

### LINKED ISSUE #6307: Visual Studio 2022 support
## What would you like to be added:
Make the VSIX extension available in Visual Studio 2022.
Now it's just not supported in the new version of VS:

image

Useful links:
- [Update a Visual Studio extension for Visual Studio 2022](https://docs.microsoft.com/en-us/visualstudio/extensibility/migration/update-visual-studio-extension?view=vs-2022)
- [Writing Visual Studio Extensions with Mads - 64bit extensions for Visual Studio 2022](https://www.youtube.com/watch?v=L8b3kFriwCs)
- [Writing Visual Studio Extensions with Mads - Splitting extensions for Visual Studio 2022](https://www.youtube.com/watch?v=-PKIPTW6km0)

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): fixes #6307, fixes https://github.com/unoplatform/uno/issues/6273, related to #5873, fixes https://github.com/unoplatform/uno/issues/6482

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

- Adds VS 2022 templates support
- Adds `dotnet new` .NET 6 Mobile templates for UWP and WinUI
- Bump to Reunion 0.8.1
- Removes dependency on System.Json package

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SolutionTemplate/UnoSolutionTemplate.net6/Mobile/UnoQuickStart.Mobile.csproj [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		<TargetFrameworks>net6.0-android;net6.0-ios;net6.0-maccatalyst</TargetFrameworks>
+		<SingleProject>true</SingleProject>
+		<OutputType>Exe</OutputType>
+
+		<RuntimeIdentifier Condition="'$(TargetFramework)' == 'net6.0-ios'">iossimulator-x64</RuntimeIdentifier>
+		<RuntimeIdentifier Condition="'$(TargetFramework)' == 'net6.0-maccatalyst'">maccatalyst-x64</RuntimeIdentifier>
+		<InvariantGlobalization Condition="'$(TargetFramework)' == 'net6.0-maccatalyst'">true</InvariantGlobalization>
+
+		<!-- Debugger workaround https://github.com/dotnet/maui-samples/blob/8aa6b8780b12e97b157514c3bdc54bb4a13001cd/HelloMacCatalyst/HelloMacCatalyst.csproj#L7 -->
+		<!-- <MtouchExtraArgs Condition="'$(TargetFramework)' == 'net6.0-maccatalyst'">$(MtouchExtraArgs) -setenv:MONO_THREADS_SUSPEND=preemptive</MtouchExtraArgs> -->
```
Comment (carldebilly) [inline src/SolutionTemplate/UnoSolutionTemplate.net6/Mobile/UnoQuickStart.Mobile.csproj:13]: ???

File: src/SolutionTemplate/UnoSolutionTemplate.net6/Skia.WPF.Host/UnoQuickStart.Skia.Wpf.Host.csproj [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+
+	<PropertyGroup>
+		<OutputType Condition="'$(Configuration)'=='Release'">WinExe</OutputType>
+		<OutputType Condition="'$(Configuration)'=='Debug'">Exe</OutputType>
+		<TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (carldebilly) [inline src/SolutionTemplate/UnoSolutionTemplate.net6/Skia.WPF.Host/UnoQuickStart.Skia.Wpf.Host.csproj:6]: Still in 3.1 ?

**Review by carldebilly — APPROVED** [review]

File: src/SolutionTemplate/UnoSolutionTemplate.net6/Mobile/UnoQuickStart.Mobile.csproj (hunk 1 above)
Comment (jeromelaban) [reply src/SolutionTemplate/UnoSolutionTemplate.net6/Mobile/UnoQuickStart.Mobile.csproj:13]: copy/paste from original samples, will be removed once original issue is fixed

File: src/SolutionTemplate/UnoSolutionTemplate.net6/Skia.WPF.Host/UnoQuickStart.Skia.Wpf.Host.csproj (hunk 2 above)
Comment (jeromelaban) [reply src/SolutionTemplate/UnoSolutionTemplate.net6/Skia.WPF.Host/UnoQuickStart.Skia.Wpf.Host.csproj:6]: Will fix in next PR

**Pull request merged**